cdc/sink/mq/dispatcher/partition/index_value.go

```diff
@@ -33,6 +36,8 @@ func NewIndexValueDispatcher() *IndexValueDispatcher {
 // DispatchRowChangedEvent returns the target partition to which
 // a row changed event should be dispatched.
 func (r *IndexValueDispatcher) DispatchRowChangedEvent(row *model.RowChangedEvent, partitionNum int32) int32 {
+	r.lock.Lock()
+	defer r.lock.Unlock()
 	r.hasher.Reset()
 	r.hasher.Write([]byte(row.Table.Schema), []byte(row.Table.Table))
```

Comment:
Does the lock protects access of `r.hasher` only?

Reply:
Yes, it protects the atomicity of these operations.